### PR TITLE
Attempt to allow coercions between SProp and sorts

### PR DESCRIPTION
**Pull request by SkySkimmer**
In order to trigger coercion we need to fail cumulativity, so this
only works with Unset Elaboration StrictProp Cumulativity

I tried locally disabling cumulativity in coercion.ml but then we lose
its use case

eg when elaborating

~~~coq
   fun (A:SProp) (P:A -> Prop) x (v:P x) => ...
~~~

`x : _ : Type` is coerced to `A : SProp` which fails

What we really need is some notion of type variable which can be
instantiated with SProp, until we have that we can let the user choose
between elaboration cumulativity (which needs less type annotations)
and coercions.

Overlay: https://github.com/LPCIC/coq-elpi/pull/85

### CODE REVIEW

**Comment by SkySkimmer** [comment]
cc @mattam82

File: pretyping/classops.ml [hunk 1]
```diff
@@ -24,6 +24,7 @@ open Mod_subst
 
 type cl_typ =
   | CL_SORT
+  | CL_SPROP
```
Comment (gares) [inline pretyping/classops.ml]: I don't get why SPROP is not part of SORT (that also makes an overlay necessary

**Comment by gares** [comment]
I find the type for classes a bit confusing, since `CL_SORT` and `CL_SPROP` seem to overlap, but actually the former does not include `sprop`. I think you should clarify that.

**Comment by SkySkimmer** [comment]
I now think we should get proper sort variables before we can do this correctly.

**Pull request closed without merging**